src/hooks/usePortfolio/useExtraTokens.ts

```diff
@@ -28,21 +29,50 @@ export default function useExtraTokens({
     [extraTokens]
   )
 
-  const onAddExtraToken = useCallback(
-    (extraToken: Token) => {
-      const { address: extraTokenAddress, name, symbol } = extraToken
-      if (extraTokens.map(({ address }: any) => address).includes(extraTokenAddress))
-        return addToast(`${name} (${symbol}) is already added to your wallet.`)
+  const checkIsTokenEligibleForAddingAsExtraToken = useCallback<
+    UsePortfolioReturnType['checkIsTokenEligibleForAddingAsExtraToken']
+  >(
+    (extraToken) => {
+      const { address, name, symbol } = extraToken
+
+      if (extraTokens.map((t) => t.address).includes(address))
+        return {
+          isEligible: false,
+          reason: `${name} (${symbol}) is already added to your wallet.`
+        }
+
       if (
         constants?.tokenList &&
         Object.values(constants.tokenList)
           .flat(1)
-          .map(({ address }: any) => address)
-          .includes(extraTokenAddress)
+          .map((t) => t.address)
+          .includes(address)
```

Comment:
@gergana95 , here maybe we should do `address.toLowerCase()` and in general - for all similar checks - we should always lowercase the incoming address input. What do you think?